File: Explorer/BlogService/Models/Blog.cs
namespace BlogService.Models;

public class Blog
{
    public int Id { get; set; }
    public required string Title { get; set; }
    public required string DescriptionMarkdown { get; set; }

    public int AuthorId { get; set; }
    public required string AuthorUsername { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public string[]? ImageUrls { get; set; }
    public ICollection<BlogComment> Comments { get; set; } = new List<BlogComment>();
    public ICollection<BlogLike> Likes { get; set; } = new List<BlogLike>();
}

File: Explorer/ExplorerFrontend/src/shared/types/blog.ts
export type CommentResponse = {
  id: number
  blogId: number
  authorId: number
  authorUsername: string
  text: string
  createdAtUtc: string
  updatedAtUtc: string
}

export type BlogResponse = {
  id: number
  title: string
  descriptionMarkdown: string
  authorId: number
  authorUsername: string
  createdAtUtc: string
  imageUrls: string[] | null
  likesCount: number
  // Follower service feed ne vraća komentare, blog servis vraća
  comments?: CommentResponse[]
}
